path_login.go

```diff
@@ -222,129 +214,100 @@ func (b *kubeAuthBackend) aliasLookahead(ctx context.Context, req *logical.Reque
 	}, nil
 }
 
+// parseAndVerifySignature parses the JWT token validating the signature against
+// any of the keys passed in.
+func (b *kubeAuthBackend) parseAndVerifySignature(token string, keys ...interface{}) (*jwt.Token, error) {
+	for i, k := range keys {
+		result, err := jwt.Parse(token, func(t *jwt.Token) (interface{}, error) {
+			return k, nil
+		})
+		if result != nil && err == nil {
+			return result, nil
+		}
+		if err != nil && strings.ToLower(err.Error()) == strings.ToLower(jwt.ErrTokenExpired.Error()) {
+			return nil, err
+		}
+		b.Logger().Debug(fmt.Sprintf("JWT signature did not validate with key %d, testing next key", i))
```

Comment:
Yes. The order should be the same order that the user configured the endpoint with when they set `pem_keys`.

Logging the key itself was not helpful, I found, since it tends to give you the public key components, and not the PEM encoding that the user passed in (probably).